code/game/mecha/mech_fabricator.dm

```diff
@@ -182,7 +182,13 @@
 	desc = initial(desc)
 
 	var/obj/item/I = new D.build_path
-	I.loc = get_step(src,SOUTH)
+	if(D.locked)
+		var/obj/item/weapon/storage/lockbox/large/L = new /obj/item/weapon/storage/lockbox/large(get_step(src,SOUTH)) //(Don't use capitals in paths, or single letters.
+		I.loc = L
+		L.name += " [initial(D.name)]"

```

Comment:
I am not sure if it will cause any problems down the line, but in game it looks better
The name becomes descriptive and explains exactly what the lockbox contains
This is a  Large lockbox Exosuit Weapon ("Ultra AC 2" LMG). It is a bulky item